ee/clickhouse/queries/paths/path_event_query.py

```diff
@@ -119,27 +119,31 @@ def _get_event_query(self) -> Tuple[str, Dict[str, Any]]:
         return "", {}
 
     def _should_query_url(self) -> bool:
-        if self._filter.target_events == [] and self._filter.custom_events == []:
-            if PAGEVIEW_EVENT not in self._filter.exclude_events:
-                return True
-        else:
-            if self._filter.include_pageviews:
-                return True
+        if (
+            self._filter.target_events == [] and self._filter.custom_events == []
```

Comment:
Q: Why not include this logic in the filter object?

Reply:
Good question! My impression is: Filter object has minimal logic (there's defaults and helpers to quickly access properties), but no business logic. It represents what the user / caller wants done.

These functions are very specific to the querying logic, deciding whether to include a query part or not, not whether the user asked for it or not. Hence opted to not put it there.

If it turns out that this is also useful elsewhere, then worth rethinking where to do this.

Reply:
In this case though the filter does _not_ contain what the user wants - we're instead embedding traits of a hypothetical UI (which does not exist yet) into "consuming" the filter object here.

I'll leave this one up to you - but if you find yourself copying logic like this a lot it's a sign of this kind of mismatch you want to isolate.

Reply:
Agree: https://github.com/PostHog/posthog/pull/5859#issuecomment-915382853

I'll most probably refactor this once we get more info about the UI and how we want to call the API.